app/src/main/java/org/ometa/lovemonster/service/ResponseParser.java

```diff
@@ -196,6 +196,12 @@ private Calendar parseDatetime(@Nullable final String datetime) {
      *      the non-blank value, or null otherwise
      */
     private String parseString(final JSONObject jsonObject, final String attributeName) {
+        // JSONObject will helpfully coerce null to "null" for us.  We don't want that.
+        // http://code.google.com/p/android/issues/detail?id=13830
+        if (jsonObject.isNull(attributeName)) {
```

Comment:
Interesting.  The issue I cited in the comment is from 2011, I wouldn't be
surprised if a newer version behaves in a sane fashion.

On Wed, Nov 25, 2015 at 11:58 PM Anthony Caliendo <notifications@github.com>
wrote:

> In app/src/main/java/org/ometa/lovemonster/service/ResponseParser.java
> <https://github.com/lovemonster/love-monster-android/pull/10#discussion_r45946320>
> :
>
> > @@ -196,6 +196,12 @@ private Calendar parseDatetime(@Nullable final String datetime) {
> >       *      the non-blank value, or null otherwise
> >       */
> >      private String parseString(final JSONObject jsonObject, final String attributeName) {
> > +        // JSONObject will helpfully coerce null to "null" for us.  We don't want that.
> > +        // http://code.google.com/p/android/issues/detail?id=13830
> > +        if (jsonObject.isNull(attributeName)) {
>
> So, I think the issue is that, to work with android unit tests, you need
> to explicitly include the json jar. I grabbed the latest version, which I
> think differs in behavior for the version included in the emulator.
>
> That would explain why the spec passes without the code change, but you
> see something different in the emulator.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lovemonster/love-monster-android/pull/10/files#r45946320>
> .
>